### PR TITLE
Add support for Puppet language

### PR DESCRIPTION
**Pull request by lcrownover**
Adding support for Puppet using their [provided language server](https://github.com/puppetlabs/puppet-editor-services)

### CODE REVIEW

File: lua/lspconfig/puppet.lua [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+local server_name = "puppet"
+local bin_name = "puppet-languageserver"
+
+configs[server_name] = {
+  default_config = {
+    cmd = {bin_name, "--stdio"};
+    filetypes = {"puppet"};
+    root_dir = util.root_pattern("manifests", ".puppet-lint.rc", "hiera.yaml", ".git");
+  };
+  -- on_new_config = function(new_config) end;
```
Comment (mjlbach) [inline lua/lspconfig/puppet.lua:13]: remove these

File: lua/lspconfig/puppet.lua [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+local util = require 'lspconfig/util'
+
+local server_name = "puppet"
+local bin_name = "puppet-languageserver"
+
+configs[server_name] = {
+  default_config = {
+    cmd = {bin_name, "--stdio"};
+    filetypes = {"puppet"};
+    root_dir = util.root_pattern("manifests", ".puppet-lint.rc", "hiera.yaml", ".git");
```
Comment (mjlbach) [inline lua/lspconfig/puppet.lua:11]: can the server also handle falling back to util.path.dirname(fname)?  see pyright

File: lua/lspconfig/puppet.lua [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  };
+  -- on_new_config = function(new_config) end;
+  -- on_attach = function(client, bufnr) end;
+  docs = {
+    description = [[
+https://github.com/puppetlabs/puppet-editor-services
```
Comment (mjlbach) [inline lua/lspconfig/puppet.lua:17]: Can you add installation instructions?

**Comment by mjlbach** [comment]
Please rebase your PR onto the latest master branch (it includes new linting checks)

**Comment by mjlbach** [comment]
Think something went wrong with the rebase :)

**Comment by lcrownover** [comment]
I think so, ha. I'll see if I can get it sorted out...

**Pull request closed without merging**